packages/strategies/stress.go

```diff
@@ -70,65 +87,220 @@ func (s *StressTest) modifyTransactions(input *types.Block, transactions types.T
 
 	result := types.Transactions{}
 
-	depositTxns := types.Transactions{}
-	userTxns := types.Transactions{}
-
-	for _, txn := range transactions {
-		if txn.IsDepositTx() {
-			depositTxns = append(depositTxns, txn)
-		} else {
-			userTxns = append(userTxns, txn)
-		}
-	}
+	depositTxns, userTxns := s.splitTxs(transactions)
 
 	currentBlockNum := input.NumberU64()
 
 	startBlockNum := s.startBlock.NumberU64()
 	// If first 10 blocks: Add one deposit transactions to send ETH to accounts
 	// Maybe if seqNum=0 (?)
 	if currentBlockNum < startBlockNum+11 {
-		i := currentBlockNum - startBlockNum - 1
-		addr := addresses[i]
+		dep := s.prepareDeposit(l1Info, currentBlockNum)
+		depositTxns = append(depositTxns, dep)
 
-		// Add deposit transaction
-		var dep types.DepositTx
+		if s.cfg.InjectERC20 && currentBlockNum == startBlockNum+1 {
+			erc20, err := s.prepareDeployErc20(input)
+			if err != nil {
+				panic(err)
+			}
 
-		source := derive.UserDepositSource{
-			L1BlockHash: l1Info.BlockHash, // l1 block info
-			LogIndex:    10_000,           // hardcode to a very large number
+			userTxns = append(types.Transactions{erc20}, userTxns...)
 		}
+	} else {
+		if s.cfg.InjectERC20 {
+			if erc20Address == (common.Address{}) {
+				s.waitForFirstBlock()
+				ctx := context.Background()
+				receipt, err := retry.Do(ctx, 3, retry.Exponential(), func() (*types.Receipt, error) {
+					return s.clients.DestNode.TransactionReceipt(ctx, deployTxHash)
```

Comment:
Instead of using a global variable specifically for `deployTxHash`, could we just use `erc20.Hash()` (`erc20` is returned from `s.prepareDeployErc20`. So could declare `var erc20 *types.Transaction` during initial `modifyTransactions` code along with other local variables [here](https://github.com/danyalprout/replayor/blob/e58edcd9a3bfe55afd64962ea767d01c36d7cab9/packages/strategies/stress.go#L86-L94)

Reply:
Same as the comment above.